src/nvim/testdir/test_random.vim

```diff
@@ -0,0 +1,51 @@
+" Tests for srand() and rand()
+
+func Test_Rand()
+  let r = srand(123456789)
+  call assert_equal([1573771921, 319883699, 2742014374, 1324369493], r)
+  call assert_equal(4284103975, rand(r))
+  call assert_equal(1001954530, rand(r))
+  call assert_equal(2701803082, rand(r))
+  call assert_equal(2658065534, rand(r))
+  call assert_equal(3104308804, rand(r))
+
+  " Nvim does not support test_settime
+  " call test_settime(12341234)
+  " let s = srand()
+  " if !has('win32') && filereadable('/dev/urandom')
+  "   " using /dev/urandom
+  "   call assert_notequal(s, srand())
+  " else
+  "   " using time()
+  "   call assert_equal(s, srand())
+  "   call test_settime(12341235)
+  "   call assert_notequal(s, srand())
+  " endif
```

Comment:
I think uncommenting the part that does not use `test_settime()` should be fine?
```suggestion
  let s = srand()
  if !has('win32') && filereadable('/dev/urandom')
    " using /dev/urandom
    call assert_notequal(s, srand())
  " else
  "   " using time()
  "   call assert_equal(s, srand())
  "   call test_settime(12341235)
  "   call assert_notequal(s, srand())
  endif
```

Reply:
I'd imagine there's a very, _very_ small chance they can be equal without the `test_settime` call above?
But it's so unlikely that it probably doesn't matter 😆

Reply:
It is also the case in Vim.

Reply:
I think `test_settime()` should not affect `/dev/urandom`?

Reply:
Oh yeah, I totally forgot about that. 🤦 🤦 🤦